lib/component.js

```diff
@@ -805,6 +843,91 @@ class Component extends WebComponent {
     return this._attrs;
   }
 
+  /**
+   * parse and validate config.params and create a param schema on the component
+   */
+  _initializeParams() {
+    // the real value for the params
+    this._params = {};
+    // maintain local validated map where all schema keys are defined
+    this._paramSchemas = {};
+    const paramSchemas = this.getConfig(`params`);
+    const defaultParams = this.getConfig(`defaultParams`);
+
+    for (let [paramName, paramSchema] of Object.entries(paramSchemas)) {
+      // convert type shorthand to object
+      if (!paramSchema.type) {
+        paramSchema = {type: paramSchema};
+      }
+
```

Comment:
removed check for `required` and `default` should not be presented same time since `require` now means the param should be declared in jsx, regardless what value it has